prboom2/src/s_sound.c

```diff
@@ -983,7 +983,7 @@ static int Raven_S_getChannel(mobj_t *origin, sfxinfo_t *sfx, sfx_params_t *para
 
   for (i = 0; i < numChannels; i++)
   {
-    if (gamestate != GS_LEVEL || origin->player)
+    if (gamestate != GS_LEVEL || origin == players[displayplayer].mo)
```

Comment:
This changes player sound behaviour in coop demos, but it probably doesn't make much difference in practice. Rewriting this file to not use ambiguous pointers is on my list, but for now this should be fine.